internalshared/configutil/listener.go

```diff
@@ -279,27 +292,28 @@ func (l *Listener) parseChrootNamespaceSettings() error {
 // The state of the listener will be modified.
 func (l *Listener) parseType(fallback string) error {
 	switch {
-	case l.Type != "":
+	case l.Type.String() != "":
 	case fallback != "":
 	default:
 		return errors.New("listener type must be specified")
 	}
 
 	// Use type if available, otherwise fall back.
-	result := l.Type
-	if result == "" {
-		result = fallback
+	rawType := l.Type
+	if rawType == "" {
+		rawType = listenerType(fallback)
 	}
-	result = strings.ToLower(result)
+
+	parsedType := listenerType(strings.ToLower(rawType.String()))
```

Comment:
I think that the **listenerType** type could use a **Parsed** method. Like this:
```
func (l listenerType) Parsed() listenerType {
    return listenerType(strings.ToLower(l.String()))
}
```
That would allow this line to become `parsedType := rawType.Parsed()`